manifests/server/role.pp

```diff
@@ -53,17 +53,17 @@
   #
   # Port, order of precedence: $port parameter, $connect_settings[PGPORT], $postgresql::server::port
   #
-  if $port != undef {
+  if $port {
```

Comment:
In this case, the priority doesn't change. `$port` is still `undef` by default. In some classes the logic seems to be broken because `$port` doesn't default to undef, but that problem exists on the main branch already.

Reply:
I think this is fine to merge because it only simplifies the code, it doesn't change the logic. but the logic is broken and should probably be checked in a dedicated issue.

Reply:
#1469 was aimed at making it more consistent. I haven't checked back on the most recent implementation, but you're right that it is inconsistent between files.